compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1903,8 +1903,8 @@ public void visit(BLangExpressionStmt exprStmtNode) {
         SymbolEnv stmtEnv = new SymbolEnv(exprStmtNode, this.env.scope);
         this.env.copyTo(stmtEnv);
         BType bType = typeChecker.checkExpr(exprStmtNode.expr, stmtEnv, symTable.noType);
-        if (bType != symTable.nilType && bType != symTable.neverType && bType != symTable.semanticError &&
-                exprStmtNode.expr.getKind() != NodeKind.FAIL) {
+        if (bType != symTable.nilType && !types.isNeverTypeOrStructureTypeWithARequiredNeverMember(bType) &&
```

Comment:
Better to have the least overhead checks first in a &&. Move  isNeverTypeOrStructureTypeWithARequiredNeverMember () to last. If the prior condition is false, rest of them won't be evaluated